File: fixtures/scale/ruby/large/lib/check_maskbyte.rb
require_relative 'maskbyte_encrypt'
require_relative 'maskbyte_decrypt'
require_relative 'corpus_mask'

def check_maskbyte
  corpus_mask.each do |text|
    sealed = maskbyte_encrypt(text, 18)
    opened = maskbyte_decrypt(sealed, 18)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_maskbyte.py
from cipherlab.ciphers.maskbyte import maskbyte_encrypt, maskbyte_decrypt
from cipherlab.specs.maskbyte_spec import maskbyte_spec
from cipherlab.corpus.corpus_mask import corpus_mask


def check_maskbyte():
    spec = maskbyte_spec()
    for text in corpus_mask():
        sealed = maskbyte_encrypt(text, spec["key"])
        opened = maskbyte_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "mask"
